### ISSUE TITLE
Authentication woes

### ISSUE THREAD

**Issue by basil-conto** [vermiculus/sx.el#342]
`sx-version`:
```
0.3 of 2018-01-28
```
`emacs-version`:
```
GNU Emacs 27.0.50 (build 4, x86_64-pc-linux-gnu, X toolkit, Xaw3d scroll bars) of 2018-02-03
```

1. `emacs -Q`
2. <kbd>M-x</kbd>`package-initialize`<kbd>RET</kbd>
3. <kbd>M-x</kbd>`sx-authenticate`<kbd>RET</kbd>
   * Expected result: browser displays page with authentication token
   * Actual result: browser reports
     ```
     `http%3A%2F%2Fseanallred.com%2Fsx.el%2Fauth%2Fauth.htm` is not a valid uri
     ```
4. <kbd>C-g</kbd>
5. <kbd>M-x</kbd>`find-function`<kbd>RET</kbd>`sx-authenticate`<kbd>RET</kbd>
6. <kbd>C-s</kbd>`sx-auth-redirect-uri`
7. <kbd>C-M-b</kbd>
8. <kbd>M-x</kbd>`raise-sexp`<kbd>RET</kbd>
9. <kbd>C-M-x</kbd>
10. <kbd>M-x</kbd>`sx-authenticate`<kbd>RET</kbd>
11. Copy token displayed in browser into minibuffer and type <kbd>RET</kbd>
12. <kbd>M-x</kbd>`toggle-debug-on-error`<kbd>`RET`</kbd>
13. <kbd>M-x</kbd>`sx-inbox`<kbd>RET</kbd>
   * Expected result: SX inbox displayed
   * Actual result:
     ```
     Debugger entered--Lisp error: (error "Request failed: (nil) [403 access_denied] \"access_token was not created with `read_inbox` scope\"")
       signal(error ("Request failed: (nil) [403 access_denied] \"access_token was not created with `read_inbox` scope\""))
       error("Request failed: (%s) [%i %s] %S" nil 403 "access_denied" "access_token was not created with `read_inbox` scope")
       sx-request-make("inbox" ((pagesize . 100) (page . 1) (filter . "abcdefghijklmnopqrs")) get sx-request-response-get-items)
       sx-method-call(inbox :keywords nil :page 1 :filter ((inbox_item\.answer_id inbox_item\.body inbox_item\.comment_id inbox_item\.creation_date inbox_item\.is_unread inbox_item\.item_type inbox_item\.link inbox_item\.question_id inbox_item\.site inbox_item\.title) (site\.logo_url site\.audience site\.icon_url site\.high_resolution_icon_url site\.site_state site\.launch_date site\.markdown_extensions site\.related_sites site\.styling)))
       sx-inbox-get(nil 1)
       #f(compiled-function (page) #<bytecode 0x12f7ed1>)(1)
       sx-question-list-refresh()
       run-hooks(tabulated-list-revert-hook)
       tabulated-list-revert()
       sx-inbox(nil)
     ```

**Comment by basil-conto** [comment]
Speedy Gonzales support team! :)

**Comment by vermiculus** [comment]
The fact that [my main project right now](https://github.com/vermiculus/magithub) is a github client certainly helps 😉

**Comment by Luis-Henriquez-Perez** [comment]
To be clear, is there an alternative way to authenticate? Or is this package unable to authenticate at by any means this time?

**Comment by basil-conto** [comment]
Here's an update of the recipe:

`emacs-version`:

    In GNU Emacs 27.0.50 (build 10, x86_64-pc-linux-gnu, X toolkit, Xaw3d scroll bars)
     of 2018-06-04 built on thunk
    Repository revision: 1dafa4a02ed45bb4d02c6dc34c55518858422088
    Windowing system distributor 'The X.Org Foundation', version 11.0.11906000
    System Description: Debian GNU/Linux buster/sid

1. `emacs -Q`
2. <kbd>M-x</kbd>`package-initialize`<kbd>RET</kbd>
3. <kbd>M-x</kbd>`sx-authenticate`<kbd>RET</kbd>
   * Expected result: browser displays page with authentication token
   * Actual result: browser reports:
     ```
     `redirect_uri` is not under the registered domain for this application
     ```

I wonder if this is caused by 46d9498e8f501697920c79437909cdddc93d5ec6?

**Comment by manandearth** [comment]
@e2m4n 

>  so that the domain part of redirect_uri was what it was before

What was it before?

**Comment by manandearth** [comment]
Oh.. found it.
You mean:
http%3A%2F%2Fseanallred.com%2Fsx.el%2Fauth%2Fauth.htm

Thanks!

**Comment by matthew-piziak** [comment]
I'm still getting "`redirect_uri` is not under the registered domain for this application`" when I run `sx-authenticate`. Is this something that hasn't been patched into the `sx` library yet?

**Comment by vermiculus** [comment]
That's likely. If someone wants to take a crack at it I'd be more than happy to review and merge a PR, but otherwise I won't likely have to investigate this with all my other commitments :frown:

**Comment by e2m4n** [comment]
The error message is from Stack Overflow, and suggests that this can only be resolved by the Stack Apps app owner by adding the new domain to the whitelist. Once that's done this issue will auto-resolve :-)

**Comment by basil-conto** [comment]
@vermiculus Thanks, everything works for me now.

**Comment by matthew-piziak** [comment]
Still broken for me. Same error: 
```
sx-request-make: Request failed: (nil) [403 access_denied] "`key` is not valid for passed `access_token`, token not found."
```

**Comment by basil-conto** [comment]
Great, I'm therefore closing this report for now.

**Comment by godfrey-cw** [comment]
Getting the same error as @matthew-piziak -- any update on this front?

**Comment by basil-conto** [comment]
> Getting the same error as @matthew-piziak -- any update on this front?

Please provide the exact steps that you are taking along with what you expect and actually see at each stage, as I think authentication has been working for everyone here for a while now.

**Comment by godfrey-cw** [comment]
Sorry, realize now my last comment wasn't very constructive.

Just followed the directions in the read me. So, 1. M-x package install ...
2. sx-authenticate 3. Copy-paste the code. 4. sx-<anything else> yields the
above mentioned error.

Further iterations of the authenticate step didn't seem to help.

One thing that made me scratch my head was the first time I went to
authenticate I was asked to create some sort of new account ... Which I did
since I didn't know how else to get through that step (even though I
already have a stack exchange account). Did I mess that step up perhaps?

Thanks for your patience all.
